src/app/features/tech-record/components/tech-record-history/tech-record-history.component.ts

```diff
@@ -1,28 +1,45 @@
-import { ChangeDetectionStrategy, ChangeDetectorRef, Component, Input } from '@angular/core';
-import { StatusCodes, TechRecordModel, VehicleTechRecordModel } from '@models/vehicle-tech-record.model';
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, Input, OnInit } from '@angular/core';
+import { StatusCodes, V3TechRecordModel } from '@models/vehicle-tech-record.model';
+import { Store } from '@ngrx/store';
 import { TechnicalRecordService } from '@services/technical-record/technical-record.service';
+import { getBySystemNumber, selectTechRecordHistory } from '@store/technical-records';
+import { Observable } from 'rxjs';
 
 @Component({
-  selector: 'app-tech-record-history[vehicle][currentTechRecord]',
+  selector: 'app-tech-record-history',
   templateUrl: './tech-record-history.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush,
   styleUrls: ['./tech-record-history.component.scss']
 })
-export class TechRecordHistoryComponent {
-  @Input() vehicle!: VehicleTechRecordModel;
-  @Input() currentTechRecord!: TechRecordModel;
+export class TechRecordHistoryComponent implements OnInit {
+  @Input() currentTechRecord?: V3TechRecordModel;
+  recordHistory$?: Observable<V3TechRecordModel[] | undefined>;
+  recordHistory?: V3TechRecordModel[];
 
   pageStart?: number;
   pageEnd?: number;
 
-  constructor(private cdr: ChangeDetectorRef, private techRecordService: TechnicalRecordService) {}
+  constructor(private cdr: ChangeDetectorRef, private techRecordService: TechnicalRecordService, private store: Store) {}
+
+  ngOnInit(): void {
+    if (this.currentTechRecord) {
+      this.store.dispatch(getBySystemNumber({ systemNumber: this.currentTechRecord?.systemNumber }));
+      this.recordHistory$ = this.store.select(selectTechRecordHistory);
+      this.recordHistory$.subscribe(records => {
+        //TODO: V3 this only sorts by created timestamp as its all that's available at the moment needs to sort by last updated when that becomes available
+        this.recordHistory = records?.sort((a, b) =>
+          a.createdTimestamp < b.createdTimestamp ? 1 : a.createdTimestamp > b.createdTimestamp ? -1 : 0
```

Comment:
isn't the createdTimestamp a string? Don't you need to convert it to a date and then compare them? Like Unix time for instance?